apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -98,13 +103,52 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                 return;
             }
 
+            var jsonStruct = new Struct();
+            jsonStruct.Fields.Add("message", Value.ForString(message));
+            jsonStruct.Fields.Add("log_name", Value.ForString(_logName));
+
+            if (GoogleLoggerScope.Current != null)
+            {
+                jsonStruct.Fields.Add("scope", Value.ForString(GoogleLoggerScope.Current.ToString()));
+            }
+
+            if (eventId.Id != 0 || eventId.Name != null)
+            {
+                var eventStruct = new Struct();
+                if (eventId.Id != 0)
+                {
+                    eventStruct.Fields.Add("id", Value.ForNumber(eventId.Id));
+                }
+                if (!string.IsNullOrWhiteSpace(eventId.Name))
+                {
+                    eventStruct.Fields.Add("name", Value.ForString(eventId.Name));
+                }
+                jsonStruct.Fields.Add("event_id", Value.ForStruct(eventStruct));
+            }
+
+            // If we have format params and its more than just the original message add them.
+            if (state is IEnumerable<KeyValuePair<string, object>> formatParams && 
```

Comment:
This will iterate over `formatParams` multiple times. Is that okay?

Reply:
It's okay but ideally we shouldn't iterate over it.

Can you elaborate on how this is happening?  Do you have suggestions to avoid this?

Reply:
Sounds good, I don't expect this to be a large list in any case.